superx/information_extractors/item_info_extractor.py

```diff
@@ -0,0 +1,248 @@
+        for zip_link in zip_links:
+            # fix zip link url if neccessary
+            if not self.current_super['zip_link_prefix'] is None:
+                zip_link = self.current_super['zip_link_prefix'] + zip_link
+
+            try:
+                request = requests.get(zip_link)
+                content = request.content
+            except Exception:
+                logging.error(f'Unable to extract from zip file with url: {zip_link}')
+            else:
+                xml_file = gzip.decompress(content).decode('utf-8')
+                # parses the xml document into a tree
+                tree = ET.fromstring(xml_file)
+                # gets child containing item information
+                items = tree.getchildren()[-1]
+                self.extract_information(items)
+
+    def extract_information(self, items):
+        """
+        This method iterates over all items in the supermarket and extracts the relevant data
+        The data is then committed into the relevant table in the data base
+        :param items: The child of the parsed xml tree containing all item information
+        """
+        item_attr_name = self.current_super['item_attr_name']
+        bIsWeighted = self.current_super['is_weighted_attr_name']
+
+        for item in items.findall(item_attr_name):
+            item_code = int(item.find('ItemCode').text)
+            name = item.find('ItemName').text
+            # exclude unwanted names from DB
+            for exclude in self.exclude_names:
+                if exclude in name:
+                    continue
+
+            quantity = Decimal(item.find('Quantity').text)
+            price = Decimal(item.find('ItemPrice').text)
+            update_date = self.standardize_date(item.find('PriceUpdateDate').text)
+            is_weighted = False
+            if item.find(bIsWeighted).text == '1':
+                is_weighted = True
+
+            unit_of_measure = self.convert_unit_name(item.find('UnitQty').text)
+            # if item already in db then continue to next item
+            if bool(Product.query.filter_by(id=item_code).first()):
+                # adding current_branch_price to the db and commit
+                current_branch_price = BranchPrice(item_code=item_code, price=price, update_date=update_date)
+                db.session.add(current_branch_price)
+                db.session.commit()
+                continue
+
+            # adding current_product to the db and commit
+            current_product = Product(id=item_code, name=name, quantity=quantity, is_weighted=is_weighted,
+                                      unit_of_measure=unit_of_measure)
+            db.session.add(current_product)
+            db.session.commit()
+            # adding current_branch_price to the db and commit
+            current_branch_price = BranchPrice(item_code=item_code, price=price, update_date=update_date)
+            db.session.add(current_branch_price)
+            db.session.commit()
```

Comment:
It seems that you didn't fully understand my previous comments on this.

Maybe try to remove the duplicate BranchPrice insert code.
You don't need to commit twice, only commit when the entire transaction is complete.
I don't see in BranchPrice where the BranchID is added.:
 ```
if not bool(Product.query.filter_by(id=item_code).first()): 
          current_product = Product(id=item_code, name=name, quantity=quantity, is_weighted=is_weighted,
                                      unit_of_measure=unit_of_measure)
          db.session.add(current_product)
            
# adding current_branch_price to the db and commit
current_branch_price = BranchPrice(item_code=item_code, price=price, update_date=update_date)
db.session.add(current_branch_price)
db.session.commit()

```

Reply:
Im working on updating this